### PR TITLE
br: handle special character in SQL

### LINKED ISSUE #37469: br: handle special character in SQL.
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

In SQL backup/restore if sk has special character (+), we will encounter the error. just like https://github.com/pingcap/br/issues/603, we handle it in binary. but the URL parsed before we replace it in SQL.

### 2. What did you expect to see? (Required)
success

### 3. What did you see instead (Required)
failed with SignatureDoesNotMatch.

### 4. What is your TiDB version? (Required)
v6.1.0

### PR DESCRIPTION
**Pull request by MoCuishle28**
Signed-off-by: Gaoming Zhang <zhanggaoming028@gmail.com>

### What problem does this PR solve?

Issue Number: close #37469

Problem Summary:
In SQL backup/restore if sk has special character (+), we will encounter the error.

### What is changed and how it works?

Replace the `'+'` to `%2B` in the storage URL.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: executor/brie.go [hunk 1]
```diff
@@ -574,3 +574,13 @@ func (gs *tidbGlueSession) UseOneShotSession(store kv.Storage, closeDomain bool,
 	// in SQL backup. we don't need to close domain.
 	return fn(gs)
 }
+
+// parseStorage parses the storage URL
+func parseStorage(storage string) (*url.URL, error) {
```
Comment (3pointer) [inline executor/brie.go:579]: can we use `ParseRawURL` directly?

Comment (MoCuishle28) [reply executor/brie.go]: Done. PTAL

Comment (MoCuishle28) [reply executor/brie.go]: I moved the unit test into `parse_test.go`.

**Review by YuJuncen — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**